libraries/chain/db_maint.cpp

```diff
@@ -925,7 +925,29 @@ void database::perform_chain_maintenance(const signed_block& next_block, const g
    // Reset all BitAsset force settlement volumes to zero
    for( const auto& d : get_index_type<asset_bitasset_data_index>().indices() )
    {
-      modify( d, [](asset_bitasset_data_object& o) { o.force_settled_volume = 0; });
+      if(head_block_time() < HARDFORK_CORE_518_TIME)
+         modify( d, [](asset_bitasset_data_object& o) { o.force_settled_volume = 0; });
+      else { // clean expired feeds
+         modify( d, [this](asset_bitasset_data_object& o) {
+            o.force_settled_volume = 0;
+
+            // Check if asset is smartcoin
+            const auto& settlement_price = o.feeds.begin()->second.second.settlement_price;
```

Comment:
This is not safe. Before dereferencing `begin()`, please check whether `feeds` is empty.

Reply:
silly me, fixed here https://github.com/bitshares/bitshares-core/pull/832/commits/f60a9530924c23b9fd0416f8653d87620e0739e5 thank you.